### DISCUSSION TITLE
Schmid Factor Calculation for Titanium

### DISCUSSION THREAD

**Discussion by huseyinhzl** [mtex-toolbox/mtex#2229] [Ask Anything]
Hello, 

I have an EBSD data of tensile test specimen after testing and I have been asked to plot and calculate area-weighted average SF of the basal and prismatic slip and average yield stress of grains. I have tried to run the codes in the documents sections, yet I could not generate anything useful. I always ended up with SF = 0 which is not realistic. I was using the below code and I believe it is not correct, so anyone can help me to guide in the coding?

Thank you so much in advance
```matlab
close all; clear variables;
startup_mtex;
set(0,'DefaultFigureWindowStyle','normal');

% Specify Crystal and Specimen Symmetries
CS = {'notIndexed',...
  crystalSymmetry('6/mmm', [3 3 4.7], 'X||a*', 'Y||b', 'Z||c*', 'mineral', 'Titanium-Hexagonal', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('m-3m', [3.3 3.3 3.3], 'mineral', 'Titanium-Cubic', 'color', [0.56 0.74 0.56]),...
  crystalSymmetry('mmm', [6.1 3.1 4.6], 'mineral', 'Ti B', 'color', [0.85 0.65 0.13])};

% Set MTEX preferences
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','intoPlane');

% Load EBSD data
pname = 'C:\Users\HIZLI-PC\Desktop\EBSD\Schmid Factor';
fname = [pname '\Amorf Horizontal HT.ctf'];
ebsd = EBSD.load(fname,CS,'interface','ctf', 'convertEuler2SpatialReferenceFrame');

%%
loadingDirection = vector3d(0, 1, 0);

%%
SlipSystems = slipSystem.basal(ebsd('Titanium-Hexagonal').CS);
sS = SlipSystems.symmetrise;
% rotate slip systems into specimen coordinates
sSLocal = grains.meanOrientation * sS

% compute Schmid factor
sigma = stressTensor.uniaxial(vector3d.Z)
SF = sS.SchmidFactor(sigma);

% take the maxium allong the rows
[SFMax,active] = max(SF,[],2);

% plot the maximum Schmid factor
plot(SFMax,'micronbar','off','linewidth',2)
mtexColorbar location southoutside
```

**Comment by Tijmenvermeij** [comment]
Try taking the absolute value of the SF, before calculating the max. The problem is that a SF can also be negative, but for slip this doesn't make sense because slip can happen in two directions.

So just replace `SF = sS.SchmidFactor(sigma);` with `SF = abs(sS.SchmidFactor(sigma));` and it should work better.

**Comment by mugikumar** [comment]
Dear Mtex user,
                  For Schmid factor analysis, is this required to do the EBSD data analysis after tensile test. if i have a EBSD data before tensile test. is this possible to get which slip system activates first. Thanks in advance

**Reply by Tijmenvermeij** [reply]
Hi,

That's more a scientific question than an Mtex question, I would say :)
1. Generally doing EBSD after a test will not get you better results in terms of SF, because at that point elastic/plastic rotations may have occurred  and then it does not represent the initial state anymore. Although you could argue that the SF may better describe new plasticity that initiates after a significant deformation
2. The SF only indicates how high the resolved shear stress is on a slip system compared to others. It does not predict anything necessarily, because stress states can vary in polycristalline microstructures, you could perhaps have different slip families with different CRSS, etc...
